src/superblock.h

```diff
@@ -0,0 +1,113 @@
+// Copyright 2018-2022 VMware, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+/*
+ * super_block.h --
+ *
+ *     This file contains a basic superblock implementation.
+ */
+
+#ifndef __SUPERBLOCK_H
+#define __SUPERBLOCK_H
+
+/* Some randomly chosen Splinter super-block checksum seed. */
+#define TRUNK_SUPER_CSUM_SEED (42)
+
+/*
+ *-----------------------------------------------------------------------------
+ * Splinter Super Block: Disk-resident structure.
+ * Super block lives on page of page type == PAGE_TYPE_SUPERBLOCK.
+ *-----------------------------------------------------------------------------
+ */
+typedef struct ONDISK trunk_super_block {
+   uint64 root_addr; // Address of the root of the trunk for the instance
+                     // referenced by this superblock.
+   uint64      meta_tail;
+   uint64      log_addr;
+   uint64      log_meta_addr;
+   uint64      timestamp;
+   bool        checkpointed;
+   bool        unmounted;
+   checksum128 checksum;
+} trunk_super_block;
+
+void
+trunk_set_super_block(trunk_handle *spl,
```

Comment:
I guess eliminate the `trunk_` prefix on these, if superblocks are going to become their own entity now?